src/rules/linalg.jl

```diff
@@ -31,7 +31,7 @@ end
 function rrule(::typeof(inv), x::AbstractArray)
     Ω = inv(x)
     m = @thunk(-Ω)
-    return Ω, Rule(ΔΩ -> m' * ΔΩ * Ω')
+    return Ω, Rule(ΔΩ -> extern(m)' * ΔΩ * Ω')
```

Comment:
Ah good catch. IIRC, this was from back when `adjoint` was defined on `t::Thunk` to be something like `@thunk(extern(t)')`.

Perhaps we should just move the adjoint into the `@thunk`, e.g.:

```julia
m = @thunk(-Ω')
return Ω, Rule(ΔΩ -> m * ΔΩ * Ω')
```

Reply:
Do we even need a `Thunk` here? What do we gain in this particular case?

Reply:
Good point. The general reason to use a `Thunk` for closed-over computations is to avoid performing the computation if it ends up being unnecessary; e.g. if a `Zero` is passed in for the differential. However, that seems pretty unlikely for a unary rule.

Reply:
Ohhh I see. `Zero` has higher precedence than `Thunk` in your monad-y dispatch list, meaning that if the stuff on the RHS of the `Thunk` is `Zero` then the `Thunk` is never materialised and we avoid the computation entirely. Have I understood this correctly? (I'm still getting too grips with this...)